internal/iam/repository_role_grant.go

```diff
@@ -494,3 +519,100 @@ func (m *MultiGrantTuple) TestStableSort() {
 	sort.Strings(gts)
 	m.Grants = strings.Join(gts, "^")
 }
+
+// grantsForUserGlobalResources returns the grants for the user for resources that can
+// only be globally scoped.
+func (r *Repository) grantsForUserGlobalResources(
+	ctx context.Context,
+	userId string,
+	res resource.Type,
+	opt ...Option,
+) (perms.GrantTuples, error) {
+	const op = "iam.(Repository).GrantsForUserGlobalResources"
+	if userId == "" {
+		return nil, errors.New(ctx, errors.InvalidParameter, op, "missing user id")
+	}
+
+	const (
+		anonUser = `where public_id in (?)`
+		authUser = `where public_id in ('u_anon', 'u_auth', ?)`
+	)
+
+	var query string
+	switch userId {
+	case globals.AnonymousUserId:
+		query = fmt.Sprintf(grantsForUserGlobalResourcesQuery, anonUser, res.String())
+	default:
+		query = fmt.Sprintf(grantsForUserGlobalResourcesQuery, authUser, res.String())
+	}
```

Comment:
See my comment above on this query, I think this can be updated to eliminate the `%s` and the use of `fmt.Sprintf` for constructing this query.